docs/express-vs-core.qmd

```diff
@@ -0,0 +1,262 @@
+The object this produces is actually straight up HTML, which is sent to the browser when it first loads the app.
+
+The server section is a function, named `server` by convention, that always takes the arguments `input`, `output`, and `session`.
+This function contains render functions and reactive functions, which are used to update the UI in response to user input.
+
+You can think of the UI section as a template, with `ui.output_xx()` calls as placeholders for outputs, and the server section as the instructions for how to populate those outputs.
+
+```{shinylive-python}
+#| standalone: true
+#| components: [editor, viewer]
+#| layout: vertical
+# Core
+from shiny import ui, render, reactive, App
+from datetime import datetime
+
+app_ui = ui.page_fixed(
+    ui.h1("Title"),
+    ui.output_code("greeting"),
+)
+
+def server(input, output, session):
+    @reactive.Calc
+    def time():
+        reactive.invalidate_later(1)
+        return datetime.now()
+
+    @render.code
+    def greeting():
+        return f"Hello, world!\nIt's currently {time()}."
+
+app = App(app_ui, server)
+```
+
+**In Shiny Express, there isn't this hard distinction** between UI and server.
+
+Instead, everything coexists as top-level code in the app file: input components, layout directives, outputs (including their rendering logic), and reactive functions.
+
+Also, you aren't forced to combine your UI into a single complex object.
+You can use multiple UI objects, and they will be combined together by the framework.
+
+```{shinylive-python}
+#| standalone: true
+#| components: [editor, viewer]
+#| layout: vertical
+# Express
+from shiny import ui, render, reactive, App
+import shiny.express
+from datetime import datetime
+
+ui.h1("Title")
+
+@reactive.Calc
+def time():
+    reactive.invalidate_later(1)
+    return datetime.now()
+
+@render.code
+def greeting():
+    return f"Hello, world!\nIt's currently {time()}."
+```
+
+Again, notice how `greeting` in this app does _not_ have a corresponding call to `output_code("greeting")`.
+This is because in Shiny Express, `render` functions automatically add an output to the page---no need to do it manually.
+
+##### Core advantages
+
+- Because the UI structure is kept separate from the server, it is easier to read, reorder, and restructure. This advantage grows as app UIs grow larger.
+- Explicit server function declaration gives us a natural place to put code that should only execute at startup (top level) versus for each session (server function body). In contrast, in Express, all of the code in the app file is executed for each session.
+
+##### Express advantages
+
+- It's nice for beginners not to have to learn about the difference between UI and server.
+- Avoids having to write code in two different places for a single output, and having to make the IDs match up.
+- No need to write nested function declarations (i.e. functions inside the server function), which can be surprising to Python programmers.
```

Comment:
```suggestion
- No need to write nested function declarations (i.e. functions inside the server function)
```

Reply:
This reads as though we shipped a bad framework and are fixing it, which I don't think is what we mean

Reply:
I don't read it that way, and I think taking off the second half makes it worse, not better. Your version makes it sound like nested function declarations are an inherent disadvantage, mine just says it's "surprising" to Python programmers.